pkg/resource/parameter_group/update_pre_build_request.go

```diff
@@ -77,3 +79,88 @@ func (rm *resourceManager) resetAllParameters(
 	}
 	return &resource{ko}, nil
 }
+
+func (rm *resourceManager) updateTags(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncTags")
+	defer func() { exit(err) }()
+
+	arn := (*string)(latest.ko.Status.ACKResourceMetadata.ARN)
+
+	toAdd, toDelete := computeTagsDelta(
+		desired.ko.Spec.Tags, latest.ko.Spec.Tags,
+	)
+
+	if len(toDelete) > 0 {
+		rlog.Debug("removing tags from parameter group", "tags", toDelete)
+		_, err = rm.sdkapi.UntagResourceWithContext(
+			ctx,
+			&svcsdk.UntagResourceInput{
+				ResourceArn: arn,
+				TagKeys:     toDelete,
+			},
+		)
+		rm.metrics.RecordAPICall("UPDATE", "UnTagResource", err)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(toAdd) > 0 {
+		rlog.Debug("adding tags to parameter group", "tags", toAdd)
+		_, err = rm.sdkapi.TagResourceWithContext(
+			ctx,
+			&svcsdk.TagResourceInput{
+				ResourceArn: arn,
+				Tags:        sdkTagsFromResourceTags(toAdd),
+			},
+		)
+		rm.metrics.RecordAPICall("UPDATE", "TagResource", err)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func computeTagsDelta(
+	desired []*svcapitypes.Tag,
+	latest []*svcapitypes.Tag,
+) (added []*svcapitypes.Tag, removed []*string) {
+	toDelete := []*string{}
+	toAdd := []*svcapitypes.Tag{}
+
+	desiredTags := map[string]string{}
+	for _, tag := range desired {
+		desiredTags[*tag.Key] = *tag.Value
```

Comment:
In case the user provides a desired tag key but no value, the code is going to panic here

Reply:
Is a tag pair that only has key but no value acceptable? Currently, I added an If condition statement to run L140 only if value is not empty. Do I need to add an error if input tag pair has no value?